### ISSUE TITLE
elemental2-dom 1.0.0 has classes that are missing in externs, even in matching closure-compiler release

### ISSUE THREAD

**Issue by niloc132** [google/elemental2#141]
Release [1.0.0](https://github.com/google/elemental2/tree/1.0.0) was tagged on Sep 19, 2019 (though the linked commit talks about 1.1.0? are we expecting a new release). That tree included this patch, committed Sep 4, 2019, and removed `NameList` integer entities:

https://github.com/google/elemental2/commit/78ad8dd1a0e5234b7cb9a7a998ed2e609e73e9bc

This patch was made because closure-compiler removed (on Sept 5, 2019) those externs:

https://github.com/google/closure-compiler/commit/fdce7045538dd1252b116c47ddc608a5b3131dd3

However, the `NameList`, `Entity`, etc types are all still included in the 1.0.0 release jars, though of course with those integer entities as doubles. I don't understand why this would be the case, since the tag diff above shows clearly that the v20190909 release of closure was used, which did not contain the NameList type: https://github.com/google/closure-compiler/blob/v20190909/externs/browser/w3c_dom1.js (you can even see that Peter's commit was the last thing to touch that file before the release).

In short, it looks like 1.0.0 was cut with an earlier version of closure-compiler, and the tagged release of closure isn't compatible when you use dependencyMode=SORT_ONLY. There may be other failure modes, that's just the main one we've run into so far.

Workaround:
To use the 1.0.0 tagged release of elemental2 with a tagged release of closure, it seems you either need to fork 20190909 and roll back that commit, or use the next previous release (I have not done this, since this may well mean using an equally old version of j2cl or introduce other surprises).

**Comment by jDramaix** [comment]
For adding more context to your issue: I guess you are using elemental2 1.0.0 in the context of j2cl and maven ? 

Elemental2 1.0.0 has be generated using closure compiler version `v20190819` and not `v20190909`. In this version Peter's commit is not in and NameList type exist: https://github.com/google/closure-compiler/blob/v20190819/externs/browser/w3c_dom3.js#L80

you can find the version of jscompiler used by following the dependencies chain:
`Elemental2-1.0.0` -> `J2CL-v20190918` -> `rules_closure-0.9.0` -> `closure compiler-v20190819`

**Comment by niloc132** [comment]
Ugh, sorry, still learning bazel. https://github.com/google/elemental2/blob/1.0.0/build_defs/repository.bzl#L6 .. which is `@com_google_closure_compiler`, not `com_google_javascript_closure_compiler`, which is where externs() pulls its rules from (as opposed to the other rules in third party, which pull from the slightly newer version). Not the first time this confusion has bitten me.
